desktop/core/src/desktop/lib/thrift_util.py

```diff
@@ -338,6 +338,22 @@ def connect_to_thrift(conf):
   if conf.transport_mode == 'http':
     if conf.use_sasl and conf.mechanism != 'PLAIN':
       mode.set_kerberos_auth(service=conf.kerberos_principal)
+
+    elif USE_THRIFT_HTTP_JWT.get():
+      from desktop.auth.backend import find_user # Cyclic dependency
+      user = find_user(conf.username)
```

Comment:
Here you find but never `rewrite_user`, check the error trace?

Reply:
Ahh, I thought once the profile has been updated/written its always with the User object. Stupid mistake! Fixing it.

Reply:
Those are brand new objects in each view/methods, like regular local variables